### PR TITLE
Ruby Group Capstone Project - Catalogue of things

### LINKED ISSUE #34: Kanban board setup
This issue has been created to provide task assignments and team information for [Ruby Group Capstone](https://github.com/Uzair-Manzoor/RubyCapstone/projects/1) project:

We are team of 3 participants working on this project:

@ismayito 
@Uzair-Manzoor 
@Ayokunnumi1

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
### We are a team of 3 participants working on this project;
@ismayito 
@Ayokunnumi1 
@Uzair-Manzoor 

## This pull request has been created against the below mentioned implementations into the project as per project requirements:

1. Created the Item class in a separate .rb file, defining properties as required, except for 1-to-many relationships.
2. Implemented the methods can_be_archived?() and move_to_archive() in the Item class.
3. Created the main.rb file as the console app entry-point and implemented the startup actions for user interaction.
4. Created the Book class in a separate .rb file.
5. Created the Label class with an association to the Item class.
6. Defined properties for Book and Label classes in the constructor.
7. Implemented the add_item method in the Label class and overrode can_be_archived?() in the Book class.
8. Added unit tests for implemented methods.
9. Provided options for listing all books, labels, and adding a book.
10. Created a schema.sql file for the books and labels tables.
11. Created the MusicAlbum class in a separate .rb file.
12. Created the Genre class with an association to the Item class.
13. Defined properties for MusicAlbum and Genre classes in the constructor.
14. Implemented the add_item method in the Genre class and overrode can_be_archived?() in the MusicAlbum class.
15. Added unit tests for implemented methods.
16. Provided options for listing all music albums, genres, and adding a music album.
17. Created a schema.sql file for the music_albums and genres tables.
18. Created the Game class in a separate .rb file.
19. Created the Author class with an association to the Item class.
20. Defined properties for Game and Author classes in the constructor.
21. Implemented the add_item method in the Author class and overrode can_be_archived?
     () in the Game class.
22. Added unit tests for implemented methods.
23. Provided options for listing games, authors, and adding a game.
24. Saved data to .json files.
25. Created a schema.sql file for the games and authors tables.
26. Recorded a presentation [Zoom](https://drive.google.com/file/d/1IcTH_GTSqDpVecmrXgxGk5VzPML29w6J/view?usp=sharing) video.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    FOREIGN KEY (genre_id) REFERENCES genres (id),
+    FOREIGN KEY (author_id) REFERENCES authors (id),
+);
+
+--create music album table--
+CREATE TABLE MUSIC_ALBUM(
+    ID INTEGER PRIMARY KEY,
+    ON_SPOTIFY BOOLEAN,
+    ARCHIVED BOOLEAN,
+    PUBLISH_DATE DATE,
```
Comment (omar-muhamad) [inline schema.sql]: - [x] the `music_album` table should have only 2 columns `on_spotify` and `id`(which is referencing the item id). Please fix that to match the given [UML diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png)

image

```suggestion
CREATE TABLE music_album(
	id INT NOT NULL,
	on_spotify BOOLEAN,
	FOREIGN KEY (id) REFERENCES item (id)
);
```

File: schema.sql [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+CREATE TABLE games (
+  id INT PRIMARY KEY,
+  item_id INT, -- A foreign key to reference the parent item table
+  title VARCHAR(255),
+  publish_date DATE,
+  multiplayer BOOLEAN,
+  last_played_at DATE,
+  archived BOOLEAN DEFAULT false,
+  author_id INT, -- Add a foreign key for the author (one-to-many relationship)
+  FOREIGN KEY (author_id) REFERENCES authors (id)
```
Comment (omar-muhamad) [inline schema.sql]: - [x] The `games` table should have only 3 columns `multiplayer`, `last_played_at`, and `id`(which is referencing the item id). Please fix that to match the given [UML diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png)

image

```suggestion
CREATE TABLE game(
	id INT NOT NULL,
	multiplayer BOOLEAN,
	last_played_at DATE,
	FOREIGN KEY (id) REFERENCES item (id)
);
```

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

### Nice job, you are almost there.

giphy (2)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- [x] Please check the inline comments below ⬇️.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md [hunk 3]
```diff
@@ -1,2 +1,144 @@
...
+<details>
+  <summary>PostgreSQL</summary>
+</details>
+
+<details>
+  <summary>WSL</summary>
+</details>
+
+### Demo Link
+- **[Video Presentation](https://drive.google.com/file/d/1IcTH_GTSqDpVecmrXgxGk5VzPML29w6J/view)**
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply README.md]: - Well resolved team

File: classes/music_album.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  attr_reader :on_spotify
+
+  def initialize(id, title, publish_date, on_spotify)
+    super(id, title, publish_date)
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    !@archived && @on_spotify
+  end
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply classes/music_album.rb]: - Well resolved team

File: spec/author_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      expect(author.items).to be_empty
+    end
+  end
+
+  describe 'The add_item method should work properly' do
+    it "Add the item to author's items collection" do
+      author.add_item(@item1)
+      expect(author.items).to contain_exactly(@item1)
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/author_spec.rb:30]: - Well resolved team

File: schema.sql (hunk 2 above)
Comment (Whoistolu) [reply schema.sql]: - Well resolved team

File: schema.sql [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+CREATE TABLE authors (
+  id INT PRIMARY KEY,
+  first_name VARCHAR(255),
+  last_name VARCHAR(255)
+);
+
+CREATE TABLE books (
+    publisher VARCHAR(100),
+    cover_state VARCHAR(100),
+);
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply schema.sql]: - Well resolved team

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**